dozer-orchestrator/src/main.rs

```diff
@@ -31,6 +35,66 @@ fn render_logo() {
     info!("\nDozer Version: {VERSION}\n");
 }
 
+#[derive(Deserialize, Debug)]
+struct DozerPackage {
+    #[serde(rename(deserialize = "latestVersion"))]
+    pub latest_version: String,
+    #[serde(rename(deserialize = "availableAssets"))]
+    pub _available_assets: Vec<String>,
+    pub link: String,
+}
+
+fn version_to_vector(version: &str) -> Vec<i32> {
+    version.split('.').map(|s| s.parse().unwrap()).collect()
+}
+
+fn compare_versions(v1: Vec<i32>, v2: Vec<i32>) -> bool {
+    for i in 0..v1.len() {
+        match v1.get(i).cmp(&v2.get(i)) {
+            Ordering::Greater => return true,
+            Ordering::Less => return false,
+            Ordering::Equal => continue,
+        }
+    }
+    false
+}
+
+async fn check_update() {
+    use std::println as info;
+    const VERSION: &str = env!("CARGO_PKG_VERSION");
+    const ARCH: &str = std::env::consts::ARCH;
+    const OS: &str = std::env::consts::OS;
+
+    loop {
+        info!("Checking for updates...");
+        let request_url = format!(
+            "https://metadata.dev.getdozer.io/?version={}&build={}&os={}",
```

Comment:
I think these are better passed as query params to get instead of us constructing message directly ?

Reply:
Makes sense, was just referring to the rust cookbook, fixed.